addons/base_automation/models/base_automation.py

```diff
@@ -156,7 +156,7 @@ def _filter_pre(self, records):
         """ Filter the records that satisfy the precondition of action ``self``. """
         if self.filter_pre_domain and records:
             domain = [('id', 'in', records.ids)] + safe_eval(self.filter_pre_domain, self._get_eval_context())
-            return records.search(domain)
+            return records.sudo().search(domain).with_user(self.env.user.id)
```

Comment:
You have to put the result back with the original environment:
```suggestion
            return records.sudo().search(domain).with_env(records.env)
```
If the original environment already had the flag `su`, the call to `with_user` will reset that flag.